### ISSUE TITLE
[How] Is it possible to anonymously authenticate?

### ISSUE THREAD

**Issue by PH0lder** [hierynomus/smbj#88]
I have a NAS share that doesn't require authentication to read from, and I can't find an obvious way to set that up using smbj.  (Well to be honest, the correct use of this library is far from obvious because there are no examples to speak of.)  I will include my own code so that maybe someone can comment if I have gone in the wrong direction.

I'd expect there to be an easy way to get an "AnonymousAuthenticationContext" or something...

```
    // This line adds the BouncyCastle security provider
    SecurityProviderAdder.addSecurityProviderIfNotAlreadyPresent();

    final SMBClient client = new SMBClient();
    try (final Connection connection = client.connect("NASHOSTNAME");)
    {
      final String username = "ME";
      final String password = "MYPASS";
      final String domain = "MYDOMAIN";
      final AuthenticationContext authContext = new AuthenticationContext(username, password.toCharArray(), domain);

      final Session session = connection.authenticate(authContext);
      final Share share = session.connectShare("Temp");
      final TreeConnect treeConnect = share.getTreeConnect();
      final String hostname = "NASHOSTNAME";
      final String shareName = "SHARENAME";
      final SmbPath smbPath = new SmbPath(hostname, shareName);
      final DiskShare ds = new DiskShare(smbPath, treeConnect);
      final ShareInfo shareInfo = ds.getShareInformation();
      System.out.println("getFreeSpace() = " + shareInfo.getFreeSpace());
      share.close();
      session.close();
    }
    catch (final Exception e)
    {
      // TODO Auto-generated catch block
      e.printStackTrace();
    }
```

**Comment by PH0lder** [comment]
I thought I'd reference #35 because it mentions examples and this code *could* be a valid example?

**Comment by pepijnve** [comment]
I don't think anonymous authentication has been implemented yet.

**Comment by hierynomus** [comment]
Have you tried just using an empty username and password?
After tweaking my Windows10 VM to allow Anonymous and Guest login, that works out of the box.

**Comment by hierynomus** [comment]
I've added two static factory methods to the AuthenticationContext in commit 3e07a43. Let me know if that works for you.

**Comment by hierynomus** [comment]
@PH0lder Have you tried whether this works? If so, I can close the issue :)

**Comment by PH0lder** [comment]
Unfortunately I am using the Maven repository, so it's not really possible for me to quickly pull your commit for a test.

**Comment by hierynomus** [comment]
@pepijnve Maybe you can verify whether _anonymous_ and _Guest_ accounts work indeed?

**Comment by hierynomus** [comment]
Released 0.1.0 which contains these fixes, closing the issue
